cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -338,13 +338,13 @@ func (ks *kube2sky) generateSRVRecord(subdomain, portSegment, recordName, cName
 }
 
 func (ks *kube2sky) addDNS(subdomain string, service *kapi.Service) error {
-	if len(service.Spec.Ports) == 0 {
-		glog.Fatalf("Unexpected service with no ports: %v", service)
-	}
 	// if ClusterIP is not set, a DNS entry should not be created
 	if !kapi.IsServiceIPSet(service) {
 		return ks.newHeadlessService(subdomain, service)
 	}
+	if len(service.Spec.Ports) == 0 {
+		glog.Fatalf("Unexpected service with no ports: %v", service)
```

Comment:
It's not possible to create a service like that with our current service validation. However creating a headless service IS currently possible and hits the fatal. I can log.Info here and return if that's preferable.